### DISCUSSION TITLE
Not understanding how to make NPCs speak dialog on click.

### DISCUSSION THREAD

**Discussion by PeachSunndae** [samolego/Taterzens#120] [Q&A]
Hello. My friend and I want to add an NPC with dialog by clicking on them. But the problem is we can't figure out anything to do with said commands. We even downloaded the world and messed with it to find out and there's just...we're so utterly lost. We've been troubleshooting for hours and nothing is working. I put this in a discussion thread because this isn't exactly an issue or bug, more just "We need more in-depth command explanations" Can anyone help? We're on 1.18.2 forge, with the forge version being 40.1.73.

**Comment by samolego** [comment]
You need to create functions which clear the commands and set new ones, then you cycle through them using clicks.
an alternative would be to use 1.19 fabric version, which has command groups (see video at https://samolego.github.io/Taterzens/1.10.1/getting_started/command_actions/creating_action/)

**Reply by PeachSunndae** [reply]
The problem with using fabric is we have a huge modpack on forge...we *can't* use fabric without starting a whole new world.
